tests/test_barostat.py

```diff
@@ -0,0 +1,224 @@
+from md.thermostat.utils import sample_velocities
+
+from timemachine.lib import LangevinIntegrator, custom_ops
+
+from functools import partial
+
+from timemachine.constants import BOLTZ, ENERGY_UNIT, DISTANCE_UNIT
+
+def test_barostat_varying_pressure():
+    temperature = 300.0 * unit.kelvin
+    initial_waterbox_width = 2.0 * unit.nanometer
+    timestep = 1.5 * unit.femtosecond
+    barostat_interval = 3
+    collision_rate = 1.0 / unit.picosecond
+    seed = 2021
+    np.random.seed(seed)
+
+    pressure = 10. * unit.atmosphere
+    mol_a = hif2a_ligand_pair.mol_a
+    ff = hif2a_ligand_pair.ff
+    complex_system, complex_coords, complex_box, complex_top = build_water_system(
+        initial_waterbox_width.value_in_unit(unit.nanometer))
+
+    min_complex_coords = minimize_host_4d([mol_a], complex_system, complex_coords, ff, complex_box)
+    afe = AbsoluteFreeEnergy(mol_a, ff)
+
+    unbound_potentials, sys_params, masses, coords = afe.prepare_host_edge(
+        ff.get_ordered_params(), complex_system, min_complex_coords
+    )
+
+
+    # get list of molecules for barostat by looking at bond table
+    harmonic_bond_potential = unbound_potentials[0]
+    bond_list = get_bond_list(harmonic_bond_potential)
+    group_indices = get_group_indices(bond_list)
+
+    trajs = []
+    volume_trajs = []
+
+    lam = 1.0
+
+    bound_potentials = []
+    for params, unbound_pot in zip(sys_params, unbound_potentials):
+        bp = unbound_pot.bind(np.asarray(params))
+        bound_potentials.append(bp)
+
+    u_impls = []
+    for bp in bound_potentials:
+        bp_impl = bp.bound_impl(precision=np.float32)
+        u_impls.append(bp_impl)
+
+    integrator = LangevinIntegrator(
+        temperature.value_in_unit(unit.kelvin),
+        timestep.value_in_unit(unit.picosecond),
+        collision_rate.value_in_unit(unit.picosecond**-1),
+        masses,
+        seed
+    )
+    integrator_impl = integrator.impl()
+
+    v_0 = sample_velocities(masses * unit.amu, temperature)
+
+    baro = custom_ops.MonteCarloBarostat(
+        coords.shape[0],
+        pressure.value_in_unit(unit.bar),
+        temperature.value_in_unit(unit.kelvin),
+        group_indices,
+        barostat_interval,
+        u_impls,
+        seed
+    )
+
+    ctxt = custom_ops.Context(coords, v_0, complex_box, integrator_impl, u_impls, barostat=baro)
+    ctxt.multiple_steps(np.ones(2000)*lam)
+    ten_atm_box = ctxt.get_box()
+    ten_atm_box_sum = np.sum(ten_atm_box)
+    # Expect the box to shrink thanks to the barostat
+    assert ten_atm_box_sum < np.sum(complex_box)
+    # Set the pressure to 1 bar
+    baro.set_pressure((1 * unit.atmosphere).value_in_unit(unit.bar))
+    # Changing the barostat interval resets the barostat step.
+    baro.set_interval(2)
+    start = time.time()
+    while time.time() - start < 15:
+        ctxt.multiple_steps(np.ones(1000)*lam)
+        atm_box = ctxt.get_box()
+        if np.sum(atm_box) > ten_atm_box_sum:
+            break
```

Comment:
Can we remove this check? (With it present, I think the assertion `ten_atm_box_sum < np.sum(atm_box)` can pass also in the case of the box size randomly fluctuating, not just in the case of the box size consistently increasing)

Reply:
Removed in 2601796